syntax = "proto3";

enum Level {
  debug = 0;
  info = 1;
  warn = 2;
  error = 3;
}

message LoggingRequest {
  string message = 1;
  Level level = 2;
  string origin = 3;
}

message LoggingResponse {
  bool logged = 1;
}

// service
service LoggingService {
  rpc Logging(LoggingRequest) returns (LoggingResponse);
}
